bin/compile/element_tree_visitor.dart

```diff
@@ -0,0 +1,134 @@
+//  Copyright 2021 Abitofevrything and others.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
```

Comment:
im not fan of linense in every file since there is license in repo. can these be removed?

Reply:
This header is required to correctly apply the license to all files:
>  To apply the Apache License to your work, attach the following boilerplate notice, with the fields enclosed by brackets "[]" replaced with your own identifying information.

Reply:
https://infra.apache.org/apply-license.html#copy-per-file its recommended but its not needed. In my opining its only add unnecessary stuff to files